widgetssdk/src/main/java/com/glia/widgets/launcher/EngagementLauncher.kt

```diff
@@ -13,45 +13,69 @@ interface EngagementLauncher {
      *
      * @param activity The Activity used to launch the chat screen
      */
-    fun startChat(activity: Activity)
+    fun startChat(activity: Activity, visitorContext: String? = null)
 
     /**
      * Starts an audio engagement.
      *
      * @param activity The Activity used to launch the call screen
      */
-    fun startAudioCall(activity: Activity)
+    fun startAudioCall(activity: Activity, visitorContext: String? = null)
 
     /**
      * Starts a video engagement.
      *
      * @param activity The Activity used to launch the call screen
      */
-    fun startVideoCall(activity: Activity)
+    fun startVideoCall(activity: Activity, visitorContext: String? = null)
 
     /**
      * Starts a secure messaging.
      *
      * @param activity The Activity used to launch the secure messaging welcome or chat screen
      */
-    fun startSecureMessaging(activity: Activity)
+    fun startSecureMessaging(activity: Activity, visitorContext: String? = null)
 }
 
 internal class EngagementLauncherImpl(private val activityLauncher: ActivityLauncher) : EngagementLauncher {
 
-    override fun startChat(activity: Activity) {
-        activityLauncher.launchChat(activity)
+    /**
+     * Starts a chat engagement.
+     *
+     * @param activity The Activity used to launch the chat screen
+     * @param visitorContext Optional visitor context id from Glia Hub
+     */
+    override fun startChat(activity: Activity, visitorContext: String?) {
+        activityLauncher.launchChat(activity, visitorContext)
```

Comment:
Why are we not updating the visitor context here instead of passing it to the activity launcher layer? This is the place where we receive it from the integrators, so the sooner we get rid of it, the better.

Reply:
Good point 👍 
Changed: 5f83329c2ff1558fcd77f7a520c209e27571beef